### PR TITLE
Replace fixed-length Bytebuffer with dynamic ByteOutputStream for serialize().

### PR DESCRIPTION
**Pull request by songy23**

### CODE REVIEW

File: core/src/test/java/com/google/instrumentation/stats/StatsContextFactoryTest.java [hunk 1]
```diff
@@ -72,41 +73,43 @@ private static StatsContext testDeserialize(InputStream inputStream) throws IOEx
...
   }
 
-  private static void encodeString(String input, ByteBuffer buffer) {
-    VarInt.putVarInt(input.length(), buffer);
+  private static void encodeString(String input, ByteArrayOutputStream byteArrayOutputStream)
+      throws IOException {
+    VarInt.putVarInt(input.length(), byteArrayOutputStream);
     for (int i = 0; i < input.length(); i++) {
-      buffer.put((byte) input.charAt(i));
+      byteArrayOutputStream.write((byte) input.charAt(i));
```
Comment (dinooliva) [inline core/src/test/java/com/google/instrumentation/stats/StatsContextFactoryTest.java]: You will need to convert to UTF-8 now due to the change in the tag specs.

Comment (songy23) [inline core/src/test/java/com/google/instrumentation/stats/StatsContextFactoryTest.java]: Updated encoding in UTF-8 and add a few tests: https://github.com/google/instrumentation-java/pull/124/commits/64d8f37c0fc6d024ae9a8b6330abf75c5c432e1c

Comment (dinooliva) [inline core/src/test/java/com/google/instrumentation/stats/StatsContextFactoryTest.java]: This code still truncates the characters.

Comment (songy23) [inline core/src/test/java/com/google/instrumentation/stats/StatsContextFactoryTest.java]: Oh forgot to update the tests. Just committed and addressed.

File: core/src/test/java/com/google/instrumentation/stats/StatsContextFactoryTest.java [hunk 2]
```diff
@@ -72,41 +72,42 @@ private static StatsContext testDeserialize(InputStream inputStream) throws IOEx
...
 
-  private static void encodeString(String input, ByteBuffer buffer) {
-    VarInt.putVarInt(input.length(), buffer);
-    for (int i = 0; i < input.length(); i++) {
-      buffer.put((byte) input.charAt(i));
-    }
+  private static void encodeString(String input, ByteArrayOutputStream byteArrayOutputStream)
+      throws IOException {
+    VarInt.putVarInt(input.length(), byteArrayOutputStream);
+    byte[] utf8ByteArray = input.getBytes("UTF-8");
```
Comment (dinooliva) [inline core/src/test/java/com/google/instrumentation/stats/StatsContextFactoryTest.java]: Having this variable is of little use, you could just inline it.

Comment (songy23) [inline core/src/test/java/com/google/instrumentation/stats/StatsContextFactoryTest.java]: Sure, just updated

**Pull request merged**